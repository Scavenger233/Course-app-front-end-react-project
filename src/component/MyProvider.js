import React, { Component } from 'react';
import AuthenticationService from '../service/AuthenticationService';

export const MContext = React.createContext();  //exporting context object
class MyProvider extends Component {

state = {isUserLoggedIn: false}
render() {
        this.state.isUserLoggedIn = AuthenticationService.isUserLoggedIn();
        return (
            <MContext.Provider value={
            {   state: this.state,
                setIsUserLoggedIn: (value) => this.setState({
                    isUserLoggedIn: value })}}>
            {this.props.children}   
            </MContext.Provider>)
    }
}

export default MyProvider