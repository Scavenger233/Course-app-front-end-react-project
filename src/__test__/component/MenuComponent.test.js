import React, { Component } from 'react';

import MenuComponent from '../.././component/MenuComponent';
import renderer from 'react-test-renderer'
import MyProvider from '../.././component/MyProvider';

import { BrowserRouter as Router} from 'react-router-dom';

test("renders correctly", async() => {
    const tree = renderer.create(<MyProvider><Router><MenuComponent /></Router></MyProvider>).toJSON();
    expect(tree).toMatchSnapshot();
});
