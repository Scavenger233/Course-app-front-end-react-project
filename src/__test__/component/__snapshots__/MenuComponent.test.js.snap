// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<header>
  <nav
    className="navbar navbar-expand-md navbar-dark bg-dark"
  >
    <div>
      <a
        className="navbar-brand"
        href=""
      >
        Main Menu
      </a>
    </div>
    <ul
      className="navbar-nav"
    >
      <li>
        <a
          className="nav-link"
          href="/courses"
          onClick={[Function]}
        >
          Courses
        </a>
      </li>
    </ul>
    <ul
      className="navbar-nav navbar-collapse justify-content-end"
    >
      <li>
        <a
          className="nav-link"
          href="/login"
          onClick={[Function]}
        >
          Login
        </a>
      </li>
    </ul>
  </nav>
</header>
`;
